src/screens/authentication/UsernameRequest/index.js

```diff
@@ -49,40 +51,94 @@ const SubmitButton = ({ disabled, onPress }) => {
 }
 
 const Icon = () => (
-  <>
-    <View style={styles.iconBackground} />
-    <SvgXml
-      width={ICON_WIDTH}
-      height={ICON_HEIGHT}
-      style={styles.icon}
-      xml={usernameIcon}
-    />
-  </>
+  <SvgXml
+    width={ICON_WIDTH}
+    height={ICON_HEIGHT}
+    style={styles.icon}
+    xml={usernameIcon}
+  />
+)
+
+const Suggestions = ({ error, suggestions, onSuggestionSelected }) => (
+  error && (
+    <View style={styles.suggestionContainer}>
+      <Button
+        onPress={() => onSuggestionSelected(suggestions[0])}
+      >
+        <SuggestionBox>
+          {suggestions[0]}
+        </SuggestionBox>
+      </Button>
+      <Button
+        onPress={() => onSuggestionSelected(suggestions[1])}
+      >
+        <SuggestionBox>
+          {suggestions[1]}
+        </SuggestionBox>
+      </Button>
+      <Button
+        onPress={() => onSuggestionSelected(suggestions[2])}
+      >
+        <SuggestionBox>
+          {suggestions[2]}
+        </SuggestionBox>
+      </Button>
+      <Button
+        onPress={() => onSuggestionSelected(suggestions[3])}
+      >
+        <SuggestionBox style={styles.lastSuggestionBox}>
+          {suggestions[3]}
+        </SuggestionBox>
+      </Button>
+    </View>
```

Comment:
I think this is fine and readable, but someone may argue this could be implemented using `map` function from array.

For example

```
suggestions.map((suggestion, index) => {
return (
 <Button onPress={() => onSuggestionSelected(suggestion)}>
  <SuggestionBox>
    {suggestion}
  </SuggestionBox>
 </Button>
 )
})
```

With a check to add the `lastSuggestionBox` style. Suggestions probably won't grow more than four elements, so I think your solution is okay, but in other cases it would be good to have the `map`, as adding more elements would "scale" automatically.